File: 백준/Silver/1940. 주몽/주몽.swift
let _ = readLine(),
    m = Int(readLine()!)!,
    numberArray = readLine()!.split(separator: " ").compactMap { Int($0) }

var numberDictionary = [Int : Int]()

for number in numberArray {
    numberDictionary[number, default: 0] += 1
}

var armorCount = 0

if m.isMultiple(of: 2) {
    armorCount += numberDictionary[m / 2, default: 0] / 2
    numberDictionary[m / 2, default: 0] = 0
}

for (key, value) in numberDictionary {
    armorCount += min(value, numberDictionary[m-key, default: 0])
    numberDictionary[key] = 0
}

print(armorCount)

File: 백준/Silver/1940. 주몽/주몽.py
from collections import defaultdict

_ = input();
m = int(input())
number_list = [int(x) for x in input().split()]

number_dictionary = defaultdict(int)

armor_count = 0

for number in number_list:
    number_dictionary[number] += 1
    
if m % 2 == 0:
    armor_count += number_dictionary[m / 2] // 2
    number_dictionary[m / 2] = 0

for number, count in dict(number_dictionary).items():
    another_number = m - number
    armor_count += min(count, number_dictionary[another_number])
    number_dictionary[number] = 0

print(armor_count)